### PR TITLE
Support preflighting the store trust, for magic nix cache

### PR DESCRIPTION
**Pull request by grahamc**
##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -551,6 +565,53 @@ export class IdsToolbox {
     return false;
   }
 
+  private async preflightNixStoreInfo(): Promise<void> {
+    this.nixStoreTrust = "unknown";
```
Comment (cole-h) [inline src/index.ts:569]: We already initialize this to "unknown", do we need to do this here?

File: src/index.ts [hunk 2]
```diff
@@ -248,6 +258,9 @@ export class IdsToolbox {
       if (!(await this.preflightRequireNix())) {
         this.recordEvent("preflight-require-nix-denied");
         return;
+      } else {
+        await this.preflightNixStoreInfo();
+        this.addFact(FACT_NIX_STORE_TRUST, this.nixStoreTrust);
```
Comment (cole-h) [inline src/index.ts:263]: Should we exit with an error / log a warning if they're not `trusted`?

File: src/index.ts (hunk 1 above)
Comment (grahamc) [reply src/index.ts:569]: nice

File: src/index.ts (hunk 2 above)
Comment (grahamc) [reply src/index.ts:263]: Most of our actions don't need trust, so I'm thinking we'll leave it up to those actions: https://github.com/DeterminateSystems/magic-nix-cache-action/pull/52

**Review by cole-h — APPROVED** [review]

**Pull request merged**